### PR TITLE
modem: hl7800: Update driver to use gpio_dt_spec

### PR DESCRIPTION
**Pull request by galak**
Simplify the implementation by using gpio_dt_spec.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

**Review by gmarull — COMMENTED** [review]
suggestion

**Review by rerickson1 — CHANGES_REQUESTED** [review]

File: drivers/modem/hl7800.c [hunk 1]
```diff
@@ -912,28 +881,21 @@ static void modem_assert_pwr_on(bool assert)
 {
 	if (assert) {
 		HL7800_IO_DBG_LOG("MDM_PWR_ON -> ASSERTED");
-		gpio_pin_set(ictx.gpio_port_dev[MDM_PWR_ON],
-			     pinconfig[MDM_PWR_ON].pin, MDM_PWR_ON_ASSERTED);
+		gpio_pin_set_dt(&hl7800_cfg.gpio[MDM_PWR_ON], MDM_PWR_ON_ASSERTED);
```
[the remark this answers is not in the document]
Comment (galak) [reply drivers/modem/hl7800.c:884]: @rerickson1 thoughts / comments on this?

[the remark this answers is not in the document]
Comment (rerickson1) [reply drivers/modem/hl7800.c:884]: We could remove all the ASSERTED/DE-ASSERTED defines and move everything to device tree as it should be.

**Review by rerickson1 — CHANGES_REQUESTED** [review]

File: boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts [hunk 2]
```diff
@@ -113,14 +113,14 @@
...
-		mdm-reset-gpios = <&gpio1 15 0>;
-		mdm-wake-gpios = <&gpio1 13 0>;
-		mdm-pwr-on-gpios = <&gpio1 2 0>;
-		mdm-fast-shutd-gpios = <&gpio1 14 0>;
-		mdm-vgpio-gpios = <&gpio1 11 0>;
-		mdm-uart-dsr-gpios = <&gpio0 25 0>;
-		mdm-uart-cts-gpios = <&gpio0 15 0>;
-		mdm-gpio6-gpios = <&gpio1 12 0>;
+		mdm-reset-gpios = <&gpio1 15 (GPIO_OUTPUT | GPIO_OPEN_DRAIN)>;
+		mdm-wake-gpios = <&gpio1 13 (GPIO_OUTPUT | GPIO_OPEN_DRAIN)>;
```
Comment (rerickson1) [inline boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts:117]: Also, what about `GPIO_ACTIVE_HIGH` or `GPIO_ACTIVE_LOW`

**Review by rerickson1 — CHANGES_REQUESTED** [review]

File: boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gmarull) [reply boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts:117]: afaik `GPIO_OUTPUT` is not set in DT, am I wrong?

Comment (galak) [reply boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts]: You tell me, the driver itself wasn't setting GPIO_ACTIVE_HIGH or GPIO_ACTIVE_LOW so I just mimic what was in the pinconfig table into the dts.

Comment (rerickson1) [reply boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts]: should be:
```
mdm-reset-gpios = <&gpio1 15 (GPIO_OPEN_DRAIN | GPIO_ACTIVE_LOW)>;
mdm-wake-gpios = <&gpio1 13 (GPIO_OPEN_SOURCE | GPIO_ACTIVE_HIGH)>;
mdm-pwr-on-gpios = <&gpio1 2 (GPIO_OPEN_DRAIN | GPIO_ACTIVE_LOW)>;
mdm-fast-shutd-gpios = <&gpio1 14 (GPIO_OPEN_DRAIN | GPIO_ACTIVE_LOW)>;
```

File: drivers/modem/hl7800.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (galak) [reply drivers/modem/hl7800.c]: Done.

File: boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts (hunk 2 above)
Comment (galak) [reply boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts]: Ok updated the dts and code.  Let me know if that all looks correct or not now.

**Review by rerickson1 — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Comment by galak** [comment]
CI failures are unrelated to this.  Merging and will look at a PR to fix the CI issue.

**Pull request merged**